src/lib/parameters/parameters.cpp

```diff
@@ -1183,89 +1170,92 @@ param_import_callback(bson_decoder_t decoder, void *private, bson_node_t node)
 	 */
 
 	switch (node->type) {
-	case BSON_INT32:
-		if (param_type(param) != PARAM_TYPE_INT32) {
-			PX4_WARN("unexpected type for %s", node->name);
-			result = 1; // just skip this entry
-			goto out;
-		}
+	case BSON_INT32: {
+			if (param_type(param) != PARAM_TYPE_INT32) {
+				PX4_WARN("unexpected type for %s", node->name);
+				result = 1; // just skip this entry
+				goto out;
+			}
 
-		i = node->i;
-		v = &i;
+			int32_t i = node->i;
+			v = &i;
```

Comment:
This is incorrect: `v` will point to invalid memory once `i` goes out of scope.

Reply:
I believe this came into place when partially syncing up shmem. These are only partially used and v can be deleted.